pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -193,14 +193,32 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 			})
 		}
 
-		// add steps for subscriptions for bundles that were added through resolution
+		// add steps for subscriptions for bundles that were added through resolution and bundles that have been marked deprecated
 		for sub := range existingSubscriptions {
-			if sub.Status.CurrentCSV == op.Name {
-				continue
+			updated := false
+			if len(op.Deprecations) == 0 && corev1.ConditionStatus(sub.Status.GetCondition(v1alpha1.SubscriptionOperatorDeprecated).Type) != corev1.ConditionUnknown {
```

Comment:
`sub.Status.GetCondition` can return `nil`, so this will panic when trying to reference `.Type`. But separately, why not remove the condition if it is `Unknown`?